debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/naming/TableNamingStrategy.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.jdbc.naming;
+
+import java.util.Map;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.debezium.bindings.kafka.KafkaDebeziumSinkRecord;
+import io.debezium.connector.jdbc.JdbcSinkConnectorConfig;
+import io.debezium.sink.SinkConnectorConfig;
+import io.debezium.sink.naming.CollectionNamingStrategy;
+
+/**
+ * A pluggable strategy contract for defining how table names are resolved from kafka records.
+ *
+ * @author Chris Cranford
+ */
+@Deprecated
+public interface TableNamingStrategy extends CollectionNamingStrategy {
+
+    /**
+     * Resolves the logical table name from the sink record.
+     *
+     * @param config sink connector configuration, should not be {@code null}
+     * @param record Kafka sink record, should not be {@code null}
+     * @return the resolved logical table name; if {@code null} the record should not be processed
+     */
+    String resolveTableName(JdbcSinkConnectorConfig config, SinkRecord record);
+
+    default String resolveCollectionName(KafkaDebeziumSinkRecord record, String collectionNameFormat) {
+        var config = new JdbcSinkConnectorConfig(Map.of(SinkConnectorConfig.COLLECTION_NAME_FORMAT, collectionNameFormat));
```

Comment:
This is close, but I worry about only passing a partial config object.

For 3.0.x, ideally, we should settle for `resolveCollectionName` with a signature that passes both the `KafkaDebeziumSinkRecord` and `SinkConnectorConfig` for now. In the default method here, we can cast it to the `JdbcSinkConnectorConfig`, and we end up with the same behavior without any potential surprises.

Then, in 3.1, we can consider reworking the signature of `CollectionNamingStrategy` only to pass the format instead if we believe we want to restrict the details the user implementation has access to.

In this case, I would mark `CollectionNamingStrategy` with the `@Incubating` annotation to identify that we can will be changing this contract in the future and that users should expect changes.

Reply:
I have added now a `TemporaryBackwardCompatibleCollectionNamingStrategyProxy` to handle the fallback properly. wdyt?

Reply:
@rk3rn3r Well, that's what I call self-descriptive naming :-D

Reply:
Agreed, but definitely a more cleaner fix, bravo :tada: